api/users.js

```diff
@@ -0,0 +1,101 @@
+const express = require('express');
+const router = express.Router();
+const jwt = require('jsonwebtoken');
+const bcrypt = require('bcrypt')
+const { JWT_SECRET } = process.env;
+const { createUser, getUser, getUserByUsername } = require('../db');
+const { requireUser } = require('./utils');
+
+
+// users/
+router.get('/', async (req, res, next) => {
+    const users = await getUser();
+    try {
+        res.send({
+            users
+        })
+    } catch (error) {
+        next(error)
+    }
+});
+
+// users/my account
+router.get('/myaccount', requireUser, async (req, res, next) => {
+            res.send({
+                ...req.user
+            })
+})
+
+// users/register
+router.post('/register', async (req, res, next) => {
+    const { username, password, name, email } = req.body
+    try {
+        const _user = await getUserByUsername(username)
+        if (_user) {
+            next({
+                name: 'UserExistsError',
+                message: 'A user by that username already exists'
+            })
+        } else if (password.length < 8) {
+            next({
+                name: 'PasswordLengthError',
+                message: 'Password must be 8 or more characters long'
+            })
+        } else if (!username || !password || !email || !name) {
+            next({
+                name: 'MissingCredentials',
+                message: 'Plese fill all required fields'
+            })
+        } else {
+            const user = await createUser({
+                username,
+                password,
+                name,
+                email
+            });
+            const token = jwt.sign({
+                id: user.id,
+                username
+            }, process.env.JWT_SECRET, {
```

Comment:
instead of taking out the id and username specifically, we can just throw our user object from line 50 in here, it's slightly more optimal since we have the data stored in that user object already anyways

`token = jwt.sign(user, process.env.JWT_SECRET, { expiresIn: '1w' })`